.github/workflows/test-backend.yml

```diff
@@ -4,34 +4,59 @@ on:
   push:
     paths:
       - "backend/**"
+      - ".github/workflows/test-backend.yml"
 
 jobs:
-  build:
+  container-job:
     runs-on: ubuntu-latest
+
+    container: python:3.12
+
+    services:
+      postgres:
+        image: postgis/postgis:16-3.4
+        env:
+          POSTGRES_SCHEME: postgresql+psycopg
+          POSTGRES_USER: postgres
+          POSTGRES_PASSWORD: postgres
```

Comment:
just saying this out loud: this is fine, bc this is an ephemeral database used only for testing.